structuretoolkit/analyse/snap.py

```
@@ -0,0 +1,397 @@
+        ), "b_sum doesn't match sum of b"
+    except AssertionError:
+        print(lmp_bsum)
+        print(lmp_barr.sum(axis=0))
+
+    lmp_dbarr = extract_compute_np(lmp, "db", 1, 2, (num_atoms, num_types, 3, n_coeff))
+    lmp_dbsum = extract_compute_np(lmp, "db_sum", 0, 1, (num_types, 3, n_coeff))
+    assert np.allclose(
+        lmp_dbsum, lmp_dbarr.sum(axis=0), rtol=1e-12, atol=1e-12
+    ), "db_sum doesn't match sum of db"
+    db_atom = np.transpose(lmp_dbarr, (0, 2, 1, 3))
+
+    lmp_vbarr = extract_compute_np(lmp, "vb", 1, 2, (num_atoms, num_types, 6, n_coeff))
+    lmp_vbsum = extract_compute_np(lmp, "vb_sum", 0, 1, (num_types, 6, n_coeff))
+    assert np.allclose(
+        lmp_vbsum, lmp_vbarr.sum(axis=0), rtol=1e-12, atol=1e-12
+    ), "vb_sum doesn't match sum of vb"
+    vb_sum = np.transpose(lmp_vbsum, (1, 0, 2)) / lmp_volume * eV_div_A3_to_bar
+
+    dbatom_shape = db_atom.shape
+    vbsum_shape = vb_sum.shape
+    a_fit = np.concatenate(
+        (
+            b_sum,
+            db_atom.reshape(
+                dbatom_shape[0] * dbatom_shape[1], dbatom_shape[2] * dbatom_shape[3]
+            ),
+            vb_sum.reshape(vbsum_shape[0] * vbsum_shape[1], vbsum_shape[2]),
+        ),
+        axis=0,
+    )
+    return np.concatenate(
+        (np.array([np.eye(a_fit.shape[0])[0]]).T, a_fit), axis=1
+    ).copy()
+
+
+def calc_snap_descriptor_derivatives(lmp, structure, bispec_options, cutoff=10.0):
+    number_coef = len(get_snap_descriptor_names(twojmax=bispec_options["twojmax"]))
+    number_species = len(set(structure.get_chemical_symbols()))
+    reset_lmp(lmp=lmp)
+    write_ase_structure(lmp=lmp, structure=structure)
+    set_potential_lmp(lmp=lmp, cutoff=cutoff)
+    set_variables(lmp, **lammps_variables(bispec_options))
+    set_computes_snappy(
+        lmp=lmp,
+        bispec_options=bispec_options,
+    )
+    try:
+        lmp.command("run 0")
+    except:
+        return np.array([])
```

Am uneasy about silencing errors like this, but not sure if lammps even gives us something useful back.

---

Unfortunately LAMMPS just crashes hard, so we need this option when we want to iterate over multiple structures.

---

@pmrv's unease is extremely warranted though, so in circumstances like this where there's actually a good reason for silencing, I suggest to add a `#comment` to explain this to future devs

---

I added a corresponding comment.